src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,294 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * Ui class handles all the interactions with user of the Duke
+ * chatbot. Ui object will print directly to the output.
+ */
+public class Ui {
+    private final String SERVANT_SPEAK;
+    private final String USER_PROMPT;
+
+    public Ui () {
+        // Initialise strings to separate messages from Duke
+        // and commands from CLI
+        this.SERVANT_SPEAK = "Duke:\n";
```

Comment:
Since this attribute is initialised for each new Ui, would it be interpreted as a constant? If it is not a constant, could the naming be improved?

Reply:
https://www.geeksforgeeks.org/final-keyword-java/

According to the link above, a final variable can be initialised in three ways:
1) when declared,
2) inside an instance initializer block or constructor, or
3) inside a static block

Do you think that it is better to just initialise these variables immediately when declared?

Reply:
I believe there is no clear guideline on this, but generally if that attribute is going to be the same for all Ui objects, my personal preference would be to initialize it when it is declared.